### PR TITLE
Do not rely on pem module for validation

### PR DESCRIPTION
**Pull request by Mitko-Kerezov**
Ping @TsvetanMilanov @rosen-vladimirov

### CODE REVIEW

File: lib/services/cloud-build-service.ts [hunk 1]
```diff
@@ -84,10 +88,31 @@ export class CloudBuildService implements ICloudBuildService {
 				this.$errors.failWithoutHelp(`The specified provision: ${iOSBuildData.pathToProvision} does not exist. Verify the location is correct.`);
 			}
 
-			let certData = this.getCertificateBase64((await this.getCertificateData(iOSBuildData.pathToCertificate, iOSBuildData.certificatePassword)).cert);
-			let provisionCertificatesBase64 = (await this.getMobileProvisionData(iOSBuildData.pathToProvision)).DeveloperCertificates.map(c => c.toString('base64'));
+			const certInfo = this.getCertificateInfo(iOSBuildData.pathToCertificate, iOSBuildData.certificatePassword);
+			const certBase64 = this.getCertificateBase64(certInfo.pemCert);
+			const provisionData = await this.getMobileProvisionData(iOSBuildData.pathToProvision);
+			const provisionCertificatesBase64 = provisionData.DeveloperCertificates.map(c => c.toString('base64'));
```
Comment (rosen-vladimirov) [inline lib/services/cloud-build-service.ts]: is there any chance `provisionData.DeveloperCertificates` to be something that's not array? For example undefined? If it's possible, please use `_.map`

File: lib/services/cloud-build-service.ts [hunk 2]
```diff
@@ -84,10 +88,31 @@ export class CloudBuildService implements ICloudBuildService {
...
+			const now = new Date();
+
+			// this.$iOSDeploymentValidatorBase
+			let appId = provisionData.Entitlements['application-identifier'];
+			provisionData.ApplicationIdentifierPrefix.forEach(prefix => {
```
Comment (rosen-vladimirov) [inline lib/services/cloud-build-service.ts]: is it possible for ApplicationIdentifier to be undefined?

File: lib/services/cloud-build-service.ts [hunk 3]
```diff
@@ -84,10 +88,31 @@ export class CloudBuildService implements ICloudBuildService {
...
 
-			let certData = this.getCertificateBase64((await this.getCertificateData(iOSBuildData.pathToCertificate, iOSBuildData.certificatePassword)).cert);
-			let provisionCertificatesBase64 = (await this.getMobileProvisionData(iOSBuildData.pathToProvision)).DeveloperCertificates.map(c => c.toString('base64'));
+			const certInfo = this.getCertificateInfo(iOSBuildData.pathToCertificate, iOSBuildData.certificatePassword);
+			const certBase64 = this.getCertificateBase64(certInfo.pemCert);
+			const provisionData = await this.getMobileProvisionData(iOSBuildData.pathToProvision);
+			const provisionCertificatesBase64 = provisionData.DeveloperCertificates.map(c => c.toString('base64'));
+			const now = new Date();
+
+			// this.$iOSDeploymentValidatorBase
```
Comment (rosen-vladimirov) [inline lib/services/cloud-build-service.ts]: why?

File: lib/services/cloud-build-service.ts [hunk 4]
```diff
@@ -84,10 +88,31 @@ export class CloudBuildService implements ICloudBuildService {
...
+			provisionData.ApplicationIdentifierPrefix.forEach(prefix => {
+				appId = appId.replace(`${prefix}.`, "");
+			});
 
-			if (!_.includes(provisionCertificatesBase64, certData)) {
+			if (certInfo.organization !== constants.APPLE_INC) {
+				this.$errors.failWithoutHelp(`The specified certificate: ${iOSBuildData.pathToCertificate} is issued by an untrusted organization. Please provide a certificate issued by ${constants.APPLE_INC}`);
+			}
+
+			if (now > provisionData.ExpirationDate) {
```
Comment (rosen-vladimirov) [inline lib/services/cloud-build-service.ts]: are both DateTimes? Are they in the same format?

File: lib/services/cloud-build-service.ts [hunk 5]
```diff
@@ -84,10 +88,31 @@ export class CloudBuildService implements ICloudBuildService {
...
+				this.$errors.failWithoutHelp(`The specified provision: ${iOSBuildData.pathToProvision} has expired.`);
+			}
+
+			if (now < certInfo.validity.notBefore || now > certInfo.validity.notAfter) {
```
Comment (rosen-vladimirov) [inline lib/services/cloud-build-service.ts]: same here - will the check work in case my culture is different

**Review by rosen-vladimirov — CHANGES_REQUESTED** [review]

File: lib/services/cloud-build-service.ts (hunk 4 above)
Comment (Mitko-Kerezov) [reply lib/services/cloud-build-service.ts]: Yes

**Review by rosen-vladimirov — APPROVED** [review]
Nice work!

**Review by TsvetanMilanov — APPROVED** [review]

**Pull request merged**